### ISSUE TITLE
[Feature Request] Add api_base parameter to OpenAIConnection

### ISSUE THREAD

**Issue by ArtyomZemlyak** [microsoft/promptflow#869]
**Is your feature request related to a problem? Please describe.**
Cant use any of initial tools for create connection to OpenAI like OpenAPI.

**Describe the solution you'd like**
Just add not required parameter to OpenAIConnection like this:
```python
    def __init__(self, api_key: str, api_base: str=None, organization: str = None, **kwargs):
        configs = {"api_base": api_base, "organization": organization}
        secrets = {"api_key": api_key}
        super().__init__(configs=configs, secrets=secrets, **kwargs)
```
And also to OpenAI Connection Schema

**Describe alternatives you've considered**

**Additional context**

**Comment by brynn-code** [comment]
Ack, we will evaluate the change cost and add the parameter support soon.

**Comment by cloga** [comment]
Hi @ArtyomZemlyak , would you elaborate more about what does api_base means in OpenAI case? Per my understand, OpenAI will have unique public api_base only? Does OpenAI also support customize the endpoint like AOAI(AzureOpenAI)?

**Comment by brynn-code** [comment]
Sorry for the late response, we have summarized the tasks to do for this issue, as there are multiple parts to update, including sdk, cli, extension, workspace portal, etc. so please expect a little long time for the full support everywhere, sdk and cli will be the first to support this.
Also, we appreciate it you could share your scenarios about this, for example a proxy before calling openai api.

**Comment by ArtyomZemlyak** [comment]
@cloga This might be suitable for local services with an OpenAI compatible API. And for it to work correctly, you need to set the api_base parameter with the url of this API.

**Comment by ArtyomZemlyak** [comment]
@brynn-code Yes, sometimes there are cases when you need to set a middleware or proxy between OpenAI and internal services

**Comment by brynn-code** [comment]
Will close this issue until we support api_base everywhere.

**Comment by brynn-code** [comment]
Update: as openai shipped stable version >=1.0.0, the apis got some breaking changes, and 'api_base' was changed to 'base_url'. Add a new pr: https://github.com/microsoft/promptflow/pull/1124 , need waiting for promptflow-tools upgrade to use openai >=1.0.0.
